Src/AutoFixtureUnitTest/MailAddressGeneratorTests.cs

```diff
@@ -87,13 +94,39 @@ public void CreateWithMailAddressRequestReturnsCorrectResultUsingLocalPartFromCo
         public void CreateWithMailAddressRequestReturnsNoSpecimenWhenContextReturnsNullLocalPart()
         {
             // Fixture setup
-            var request = typeof(MailAddress);         
+            var request = typeof(MailAddress);
+            var anonymousDomainName = new DomainName(Guid.NewGuid().ToString());
+
+            var context = new DelegatingSpecimenContext()
+            {
+                OnResolve = r =>
+                {
+                    Assert.True(typeof(EmailAddressLocalPart).Equals(r) || typeof(DomainName).Equals(r));
+                    return typeof(DomainName).Equals(r) ? anonymousDomainName : null;
+                }
+            };
+            var sut = new MailAddressGenerator();
+            // Exercise system
+            var expectedResult = new NoSpecimen(request);

```

Comment:
Indeed, I also found it weird that it was part of that phase. It was already like that, but now that I've looked around elsewhere in the code, I see that's normally part of the assert phase.

Frankly, I usually make my expected outcome part of the arrange phase, since it's very often derived from the input values, and then it closer to the setup of the rest of the fixture, and you only need to do actual assert calls for the assert phase. However, after looking around, I'd say it's part of the assert phase, so I've moved it there.